src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -630,41 +630,44 @@ private bool UseSiteError()
                 return false;
             }
 
-            Symbol symbol = staticInstanceMismatch.Member;
-
-            // Certain compiler-generated invocations produce custom diagnostics.
-            if (receiverOpt?.Kind == BoundKind.QueryClause)
-            {
-                // Could not find an implementation of the query pattern for source type '{0}'.  '{1}' not found.
-                diagnostics.Add(ErrorCode.ERR_QueryNoProvider, location, receiverOpt.Type, symbol.Name);
-            }
-            else if (binder.Flags.Includes(BinderFlags.CollectionInitializerAddMethod))
-            {
-                diagnostics.Add(ErrorCode.ERR_InitializerAddHasWrongSignature, location, symbol);
-            }
-            else if (nodeOpt?.Kind() == SyntaxKind.AwaitExpression && symbol.Name == WellKnownMemberNames.GetAwaiter)
-            {
-                diagnostics.Add(ErrorCode.ERR_BadAwaitArg, location, receiverOpt.Type);
-            }
-            else if (delegateOrFunctionPointerType is FunctionPointerTypeSymbol)
-            {
-                diagnostics.Add(ErrorCode.ERR_FuncPtrMethMustBeStatic, location, symbol);
-            }
-            else
+            if (receiverOpt?.HasErrors != true)
```

Comment:
>receiverOpt?.HasErrors != true [](http://example.com/codeflow?start=16&length=30)

What is the impact of this change? Does it improve errors or does it affect non-error scenarios?

Reply:
> What is the impact of this change? Does it improve errors or does it affect non-error scenarios?

It reduces the noise in error scenarios. See the only test in QueryTests.cs, for example. However, the primary goal was to suppress cascading diagnostics on a BoundBadExpression that we create in ```ConstructBoundMemberGroupAndReportOmittedTypeArguments``` due to an ambiguity.